pkgs/development/python-modules/Theano/compiler.patch

```diff
@@ -0,0 +1,16 @@
+--- Theano-src/theano/configdefaults.py_	2017-06-14 14:12:00.000000000 -0400
++++ Theano-src/theano/configdefaults.py	2017-06-14 14:12:44.000000000 -0400
+@@ -380,11 +380,11 @@
+             'FAST_COMPILE', 'PROFILE_MODE', 'DEBUG_MODE'),
+     in_c_key=False)
+ 
+-param = "g++"
++param = "@g++@"
+ 
+ # Test whether or not g++ is present: disable C code if it is not.
+ try:
+-    rc = call_subprocess_Popen(['g++', '-v'])
++    rc = call_subprocess_Popen(['@g++@', '-v'])
```

Comment:
[Here](https://github.com/Theano/Theano/blob/0.8.X/theano/configdefaults.py#L383) is a link to the relevant bit in the patched file.  As you can see, they treat the Mac case specially by trying to run `clang++`.

Perhaps the patch should delete this compiler determination mess and replace it with

```python
param = "@compiler@"
```

to be substituted at build time by according to the Nix expression.